### ISSUE TITLE
Refactor Settings and AccountSettings

### ISSUE THREAD

**Issue by mememori** [HackerExperience/Helix#11]
As it is now, AccountSettings is an EAV and that (including the fact that each
setting value is a row/record) doesn't give us any advantage (actually it is the
opposite).

To fix this, remove `Setting`, make the possible settings (and their default
values) be a compiled map

Drop everything about `AccountSetting` and make it simply store a JSON linked to
the account.

When returning the user settings, simply merge it over the compiled settings map
to get the final map with both defaults and custom values

Ideally that compiled settings map could be an Ecto embedded schema, that way it
would be very easy to validate the input values and it could still easily be
used the way we expect it to

```elixir
defmodule Settings do
  @primary_key false
  embedded_schema do
    field :is_beta, :boolean,
      default: false
  end
end
```

Note that in this case i didn't put the default value in the embedded schema,
there's a reason for this: we might opt to create a special type for it that
strips it from struct metadata and from nil fields (thus avoiding saving the
defaults on the database) the advantage of this approach is saving database
store (well, it doesn't matter that much) and that a change on the default value
would affect every account that doesn't has a custom value for such setting
(while storing the record with the default value would make it a custom value)

AccountSettings controller should implement:
```elixir
Model.AccountSettings.__schema__ :: %AccountSettings{account_id: HELL.PK.t, settings: Settings.t}
Model.AccountSettings.settings(AccountSettings.t) :: Settings.t

# AccountController
get_settings(Account.t) :: Settings.t
put_settings(Account.t, settings :: map) :: {:ok, Settings.t} | {:error, reason :: term}
```

**Comment by kress95** [comment]
Why `AccountSettings` controller should implement `Model.AccountSettings.__schema__`?

I think I didn't get it...
What is `AccountSettings.t`, I believe that we just have `Settings.t`

**Comment by mememori** [comment]
that means that account settings is an ecto model and that it should be typed

**Comment by kress95** [comment]
Problems found while deving this:

I'm probably already 2~3 hours late on this, getting the embed to work got much time from me as I didn't understood how it worked at first.

Sadly my testing method focused the wrong spot again, and, after some testing I finally started to test it right and identify the problems, it's done, but no automated tests yet.

I just perceived that one of my main problems is being too optimistic about the pure parts of my code and starting to test things from above of it (ignoring possible problems on the pure parts).

I'll try clear the code some more and just add some more tests before sending the PR.
